playbooks/common/openshift-cluster/additional_config.yml

```diff
@@ -54,3 +54,5 @@
   - role: openshift_registry
     registry_volume_claim: "{{ openshift.hosted.registry.storage.volume.name }}-claim"
     when: deploy_infra | bool and attach_registry_volume | bool
+  - role: openshift_metrics
+    when: use_openshift_metrics | bool
```

Comment:
To integrate metrics with persistent storage, take a look at the variables we can set in the [BYO inventories](https://github.com/openshift/openshift-ansible/blob/master/inventory/byo/hosts.ose.example#L224-L256) and also take a look at this PR https://github.com/openshift/openshift-ansible/pull/1173.

We would want to provide a set of metrics variables in the example inventories similar to the registry variables. If present in the inventory, anything beginning with `openshift_hosted*` variable is persisted by [this call to openshift_facts](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_common/tasks/main.yml#L41-L46). [Defaults for these variables](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_facts/library/openshift_facts.py#L1332-L1348) are set in `openshift_facts`.


For example,

```
openshift_hosted_metrics_storage_kind=nfs
openshift_hosted_metrics_storage_nfs_directory=/exports
openshift_hosted_metrics_storage_volume_name=metrics
openshift_hosted_metrics_storage_access_modes=['ReadWriteMany']
```

These variables are persisted in the facts dictionary like this:

```
facts['openshift']['hosted']['metrics']['storage']['kind'] = 'nfs'
facts['openshift']['hosted']['metrics']['storage']['nfs']['directory'] = '/exports'
...
```

An nfs host is configured via the `openshift_storage_nfs` role which makes use of these variables to create volumes. This role currently only creates a registry volume and would need to be extended to create a metrics volume.

In the [additional cluster config playbook](https://github.com/openshift/openshift-ansible/blob/master/playbooks/common/openshift-cluster/additional_config.yml), we process hostvars with `oo_persistent_volumes` and `oo_persistent_volume_claims` filters to return a list of persistent volumes and claims to create. Both filters are component agnostic and look at `openshift_hosted*` persisted facts to determine which volumes and claims to make. Using the variables above, we'd end up with a `metrics-volume` PV and `metrics-claim` PVC and we can use these when deploying metrics.